### DISCUSSION TITLE
Will there be performance issues if the same atom is subscribed to by multiple subcomponents?

### DISCUSSION THREAD

**Discussion by iambinlin** [pmndrs/jotai#2207] [Q&A] [answered]
Hi，dear `dai-shi` @dai-shi 

![image](https://github.com/pmndrs/jotai/assets/39019913/700f44cb-6876-4fc1-8d1a-1ac42652090d)

## Q
Area `A`'s atom data will there be any performance issues if subscribed to by `D`, `F` and `E` at the same time?
**code**

```ts
useEffect(() => {
        const unSubscribe = pageStore.sub(topStore, () => {
            console.log("updated", pageStore.get(topStore))
        })
        return () => {
            unSubscribe()
        }
    }, [])
```

**Background**

 In my project, in a complex page, I need to use atom to achieve UI = f(state), but in the final submit button, I need to collect all the `atom` data. Since the data collection for submission action is unrelated to component rendering, I don't need `useAtomValue`, and I want to take advantage of the ability of `subscribe` To place it in a constant `ref`. I have been struggling for a long time because I am not familiar with related content, so I have no choice but to ask you for help. Sorry for disturbing you late at night.I would like to ask if this is correct or if you have any better suggestions.I will be very grateful.

**Comment by dai-shi** [comment, marked as answer]
If I understand your use case correctly, the recommended way is to create a write-only atom for `A` or use `useAtomCallback`.
Using `.sub()` isn't very idiomatic in this case, as it requires the explicit "store" instance.

**Reply by iambinlin** [reply]
I remember we talked about it before, in order to use `useAtomCallback`, you must use `useAtom` or `useAtomValue` to subscribe and update in the destination component so that you can get the latest value. But this will cause components D, E and F to rerender even though they don't actually need UI refresh. I just want a transient value.

**Reply by iambinlin** [reply]
I don't really want the component that takes the latest value to rerender on the scene.

**Reply by dai-shi** [reply]
No, it doesn't re-render. You don't use useAtomValue with useAtomCallback.

**Reply by iambinlin** [reply]
For example if I set atom in Component A and useAtomValue in Component B, can I get the latest value by using useAtomCallback in Components D, E and F?
In real practice, I found that sometimes I can't get it, maybe there's something wrong with my writing.

**Reply by dai-shi** [reply]
Yes, you can. I'd suggest to try creating a very small example with https://csb.jotai.org or something.

**Reply by iambinlin** [reply]
Later on I will restore my scene, thank you for your patience in answering.
